### ISSUE TITLE
Startup dB checks getting slow 

### ISSUE THREAD

**Issue by FrigoEU** [urweb/urweb#230]
Hi,

When an urweb server starts, multiple queries get executed per table to check if the database conforms to what we expect of it. This is great, but it has been causing me some trouble the more tables get added to the project. 

For me specifically this is becoming a serious issue because I run many urweb processes (one per tenant) on a fairly modest server. The startup time is reaching around 2 minutes, but once they're up and running the server is more than fast enough. 

The whole process is quite inefficient. It probably never was a problem for anybody, but I plan to fix this. In short, instead of doing 2 or 3 queries per table, I plan to just run one query: get all the Ur tables, details about the columns, etc, all in one go. I'll be doing this just for postgres for now since that's what I use.

I just wanted to check here if this is the right approach or if I should be looking a different direction.

**Comment by achlipala** [comment]
Wow, I'm surprised it got to be an issue!  How many tables are in the app with the performance trouble?

**Comment by FrigoEU** [comment]
There's 80 tables in the app at the moment. There's 55 tenants on my (small) server at the moment. I see that we're running 3 queries per table. In total, a restart of all the tenants (every time I release a new version for example) takes about 2.5 minutes.

**Comment by ashalkhakov** [comment]
You could also move this to a different binary. Or perhaps, add a command-line switch to differentiate between "normal mode" vs "post-deployment checks". That way you'd only need to check table schema once per deployment (and perhaps not even in your production environment!).

**Comment by FrigoEU** [comment]
Yeah, I thought about some other solutions. But I really think the checks are super valuable, and the queries can be much faster, so we can have both the functionality and the speed :).

For my own reference, it’s this function: https://github.com/urweb/urweb/blob/2bc06e9fee6d0326bec66a205caf10b84bb461b7/src/postgres.sml#L63
